swarm/api/http/server_test.go

```diff
@@ -892,3 +894,268 @@ func TestMethodsNotAllowed(t *testing.T) {
+				t.Fatalf("expected %s %s to have %s within HTTP response body but it didn't", testCase.method, testCase.uri, testCase.assertResponseBody)
+			}
+		})
+	}
+}
+
+func TestModify(t *testing.T) {
+	verbose := false
+	// Setup Swarm and upload a test file to it
+	srv := testutil.NewTestSwarmServer(t, serverFunc)
+	defer srv.Close()
+
+	swarmClient := swarm.NewClient(srv.URL)
+	data := []byte("data")
+	file := &swarm.File{
+		ReadCloser: ioutil.NopCloser(bytes.NewReader(data)),
+		ManifestEntry: api.ManifestEntry{
+			Path:        "",
+			ContentType: "text/plain",
+			Size:        int64(len(data)),
+		},
+	}
+
+	hash, err := swarmClient.Upload(file, "", false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// PUT/PATCH bzz:/hash -> 405 Method Not Allowed
+	putPatchHash := func() {
+		url := fmt.Sprintf("%s/bzz:/%s", srv.URL, hash)
+
+		res, _ := httpDo("PUT", url, nil, nil, verbose, t)
+
+		if res.StatusCode != 405 {
+			t.Fatal("expected PUT bzz:/hash to return a 405 but it didn't")
+		}
+
+		res, _ = httpDo("PATCH", url, nil, nil, verbose, t)
+
+		if res.StatusCode != 405 {
+			t.Fatal("expected PUT bzz:/hash to return a 405 but it didn't")
+		}
+	}
+	putPatchHash()
+
+	// DELETE bzz:/hash -> 200 OK
+	deleteHash := func() {
+		url := fmt.Sprintf("%s/bzz:/%s", srv.URL, hash)
+
+		res, body := httpDo("DELETE", url, nil, nil, verbose, t)
+
+		if res.StatusCode != 200 {
+			t.Fatal("expected DELETE bzz:/hash to return 200 but it didn't")
+		}
+		if body != "8b634aea26eec353ac0ecbec20c94f44d6f8d11f38d4578a4c207a84c74ef731" {
+			t.Fatal("After DELETE bzz:/hash, the manifest's hash is not the value we expected")
+		}
+	}
+	deleteHash()
+
+	// POST bzz-raw:/ "POSTdata" -> 200 OK
+	postData := func() {
+		url := fmt.Sprintf("%s/bzz-raw:/", srv.URL)
+		buf := bytes.NewReader([]byte("POSTdata"))
+
+		res, hash := httpDo("POST", url, buf, nil, verbose, t)
+
+		if res.StatusCode != 200 {
+			t.Fatal("expected POST bzz-raw:/ to return 200 but it didn't")
+		}
+		// Try downloading what we just uploaded
+		url = fmt.Sprintf("%s/bzz-raw:/%s", srv.URL, hash)
+
+		res, body := httpDo("GET", url, nil, nil, verbose, t)
+		if body != "POSTdata" {
+			t.Fatalf("expected %s to be 'POSTdata' but it wasn't", hash)
+		}
+	}
+	postData()
+
+	// PUT bzz-raw:/hash -> 405 Method Not Allowed
+	putHash := func() {
+		url := fmt.Sprintf("%s/bzz-raw:/%s", srv.URL, hash)
+
+		res, _ := httpDo("PUT", url, nil, nil, verbose, t)
+
+		if res.StatusCode != 405 {
+			t.Fatal("expected PUT bzz-raw:/hash to return 405 but it didn't")
+		}
+	}
+	putHash()
+
+	// POST bzz-raw:/encrypt
+	postBzzRawEncrypt := func() {
+		url := fmt.Sprintf("%s/bzz-raw:/encrypt", srv.URL)
+		buf := bytes.NewReader([]byte("POSTdata"))
+
+		res, body := httpDo("POST", url, buf, nil, verbose, t)
+
+		if res.StatusCode != 200 {
+			t.Fatalf("expected POST bzz-raw:/encrypt to return 200, but got %d", res.StatusCode)
+		}
+		if len(body) != 128 {
```

Comment:
you could hypothetically add a header matcher too (a field in the struct that has a `map[string]string` and checks the headers on the response. then you could match that header `Content-Length` equals `128` :)